interpreter/cling/lib/Interpreter/IncrementalParser.cpp

```diff
@@ -179,10 +181,49 @@ namespace {
   };
 } // unnamed namespace
 
+static void HandlePlugins(CompilerInstance& CI,
+                         std::vector<std::unique_ptr<ASTConsumer>>& Consumers) {
+  // Copied from Frontend/FrontendAction.cpp.
+  // FIXME: Remove when we switch to the new cling driver.
+
+  // If the FrontendPluginRegistry has plugins before loading any shared library
+  // this means we have linked our plugins. This is useful when cling runs in
+  // embedded mode (in a shared library). This is the only feasible way to have
+  // plugins if cling is in a single shared library which is dlopen-ed with
+  // RTLD_LOCAL. In that situation plugins can still find the cling, clang and
+  // llvm symbols opened with local visibility.
+  if (FrontendPluginRegistry::begin() == FrontendPluginRegistry::end())
+    for (const std::string& Path : CI.getFrontendOpts().Plugins) {
+      std::string Err;
+      if (llvm::sys::DynamicLibrary::LoadLibraryPermanently(Path.c_str(), &Err))
+        CI.getDiagnostics().Report(clang::diag::err_fe_unable_to_load_plugin)
+          << Path << Err;
+  }
+
+  // If there are no registered plugins we don't need to wrap the consumer
+  if (FrontendPluginRegistry::begin() != FrontendPluginRegistry::end())
+    for (auto it = clang::FrontendPluginRegistry::begin(),
+              ie = clang::FrontendPluginRegistry::end();
+              it != ie; ++it) {
+      std::unique_ptr<clang::PluginASTAction> P(it->instantiate());
+
+      PluginASTAction::ActionType PluginActionType = P->getActionType();
+      assert(PluginActionType != clang::PluginASTAction::ReplaceAction);
+
+      if (P->ParseArgs(CI, CI.getFrontendOpts().PluginArgs[it->getName()])) {
+        std::unique_ptr<ASTConsumer> PluginConsumer
+          = P->CreateASTConsumer(CI, /*InputFile*/ "");
+        if (PluginActionType == clang::PluginASTAction::AddBeforeMainAction)
+          Consumers.insert(Consumers.begin(), std::move(PluginConsumer));
+        else
+          Consumers.push_back(std::move(PluginConsumer));
+      }
+    }
+}
+
 namespace cling {
   IncrementalParser::IncrementalParser(Interpreter* interp, const char* llvmdir)
-      : m_Interpreter(interp),
-        m_ModuleNo(0) {
+      : m_Interpreter(interp), m_ModuleNo(0), m_CodeGen(nullptr) {
```

Comment:
Can you convert `m_ModuleNo(0)` into an in-class initializer, just like `m_CodeGen`, instead of adding it here?